pkg/console/controllers/advancedclustermanagement/managedclusteroauthclients/controller.go

```diff
@@ -0,0 +1,196 @@
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	// openshift
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configclientv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	oauthv1client "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	operatorinformersv1 "github.com/openshift/client-go/operator/informers/externalversions/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	clusterclientv1 "open-cluster-management.io/api/client/cluster/clientset/versioned/typed/cluster/v1"
+	workclientv1 "open-cluster-management.io/api/client/work/clientset/versioned/typed/work/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+
+	//subresources
+	manifestworksub "github.com/openshift/console-operator/pkg/console/subresource/manifestwork"
+	oauthsub "github.com/openshift/console-operator/pkg/console/subresource/oauthclient"
+
+	// console-operator
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+	"github.com/openshift/console-operator/pkg/console/status"
+)
+
+const ConditionPrefix string = "ManagedClusterOauthClientSync"
+
+type Controller struct {
+	operatorClient       v1helpers.OperatorClient
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	oauthClient          oauthv1client.OAuthClientsGetter
+	managedClusterClient clusterclientv1.ManagedClustersGetter
+	workClient           workclientv1.ManifestWorksGetter
+}
+
+func NewManagedClusterOAuthClientController(
+	// top level config
+	configClient configclientv1.ConfigV1Interface,
+	configInformer configinformer.SharedInformerFactory,
+
+	// clients
+	operatorClient v1helpers.OperatorClient,
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	managedClusterClient clusterclientv1.ManagedClustersGetter,
+	oauthClient oauthv1client.OAuthClientsGetter,
+	workClient workclientv1.ManifestWorksGetter,
+
+	// informers
+	operatorConfigInformer operatorinformersv1.ConsoleInformer,
+
+	// events
+	recorder events.Recorder,
+) factory.Controller {
+	ctrl := &Controller{
+		operatorClient:       operatorClient,
+		operatorConfigClient: operatorConfigClient,
+		managedClusterClient: managedClusterClient,
+		oauthClient:          oauthClient,
+		workClient:           workClient,
+	}
+
+	configV1Informers := configInformer.Config().V1()
+
+	return factory.New().
+		WithFilteredEventsInformers( // configs
+			util.IncludeNamesFilter(api.ConfigResourceName),
+			configV1Informers.Consoles().Informer(),
+			operatorConfigInformer.Informer(),
+		).
+		ResyncEvery(1*time.Minute).
```

Comment:
shouldnt we also add informer for the oauthClient ? Or we dont need to care if wheather it will get updated ?